### ISSUE TITLE
JS requires CSP UNSAFE_EVAL

### ISSUE THREAD

**Issue by AntSmith60** [nanostudio-org/nanogallery2#305]
**Describe the bug**
I have implemented nanogallery2 on my website and I have had to add 'unsafe_eval' to the script-src Content Security Policy, else jquery.nanogallery2.js throws an "Uncaught EvalError: call to Function() blocked by CSP" at line 11525 [var root = this || Function('return this')();]

without adding unsafe_eval to the CSP the above exception is thrown and the gallery views do not render.

**To Reproduce**
Add header "Content-Security-Policy: script-src 'self' https://cdn.jsdelivr.net/;" to webpage integrating nanogallery2

**Expected behavior**
nanogallery2  removes need to vreate the unsafe_eval security hole.

**CODEPEN**
n/a

**Screenshots**
n/a

**Desktop (please complete the following information):**
Windows 10
Tested with FireFox 82.0.3 (64-bit) - but is probably not browser specific

**Smartphone (please complete the following information):**
n/a

**Additional context**
n/a

**Comment by Kris-B** [comment]
The code causing this issue is part of the Shifty library which is included in nanogallery2.

To fix it, the line:
```
  var root = this || Function('return this')();
```
needs to be replaced by:
```
  const root = typeof window !== 'undefined' ? window : global
```

The fix will be included in the next release.

**Comment by Kris-B** [comment]
Can you please confirm that this fix is working ?

**Comment by AntSmith60** [comment]
Hi Kris
I've downloaded v3.0.4 and replaced that line as indicated and now I am able to remove unsafe-eval from my CSP - so yes I can confirm that the fix works.Many thanks
Ant

**Comment by Kris-B** [comment]
Thanks for your feedback.
Fixed in v3.0.5
